server.go

```diff
@@ -8,18 +8,69 @@ import (
 
 	"github.com/tus/tusd/pkg/filestore"
 	tusd "github.com/tus/tusd/pkg/handler"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/jwtauth"
+
+	_ "github.com/joho/godotenv/autoload"
 )
 
 var dir = flag.String("dir", "tusd-files", "where the uploaded files should be stored")
 
+var tokenAuth *jwtauth.JWTAuth
+
+func hello(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "Hello")
+}
+
+func init() {
+	secret := os.Getenv("JWTAuthSecret")
+	tokenAuth = jwtauth.New("HS256", []byte(secret), nil)
+	// For debugging/example purposes, we generate and print
+	// a sample jwt token with claims `user_id:123` here:
+	// Don't forget to import jwt "github.com/dgrijalva/jwt-go"
+	// _, tokenString, _ := tokenAuth.Encode(jwt.MapClaims{"user_id": 123})
+	// fmt.Printf("DEBUG: a sample jwt is %s\n\n", tokenString) */
+}
+
 func main() {
 	flag.Parse()
 
 	_, err := os.Stat(*dir)
 	if os.IsNotExist(err) {
 		err = os.MkdirAll(*dir, 0777)
 	}
+	err = http.ListenAndServe(":8080", router())
+	if err != nil {
+		panic(fmt.Errorf("Unable to listen: %s", err))
+	}
+}
 
+func router() http.Handler {
+	r := chi.NewRouter()
+
+	//Protected routes
+	r.Group(func(r chi.Router) {
+		// Seek, verify and validate JWT tokens
+		r.Use(jwtauth.Verifier(tokenAuth))
+
+		// Handle valid / invalid tokens. In this example, we use
+		// the provided authenticator middleware, but you can write your
+		// own very easily, look at the Authenticator method in jwtauth.go
+		// and tweak it, its not scary.
+		r.Use(jwtauth.Authenticator)
+		r.Get("/files/", http.StripPrefix("/files/", tusdHandler()).(http.HandlerFunc))
```

Comment:
Casting a handler to a `http.HandlerFunc` will work but it's ugly. You should use a method that allows registering a `http.Handler` to the router instead.

An `http.HandlerFunc` **is** an `http.Handler` because it implements its interface. But any `http.Handler` isn't guaranteed to be a `http.HandlerFunc`, it could be any type implementing the `Handler` interface.